aten/src/THC/THCBlas.cu

```diff
@@ -374,6 +378,25 @@ void THCudaBlas_HgemmStridedBatched(THCState *state, char transa, char transb, i
 }
 #endif
 
+#ifdef __HIP_PLATFORM_HCC__
+void THCudaBlas_SgemmBatched(THCState *state, char transa, char transb, int64_t m, int64_t n, int64_t k,
+                             float alpha, const float *a[], int64_t lda, const float *b[], int64_t ldb,
+                             float beta, float *c[], int64_t ldc, int64_t batchCount)
+{
+  if( (m >= INT_MAX) || (n >= INT_MAX) || (k >= INT_MAX) || (lda >= INT_MAX)  || (ldb >= INT_MAX) || (ldc >= INT_MAX) || (batchCount >= INT_MAX) )
+  {
+    THError("Cublas_SgemmBatched only supports m, n, k, lda, ldb, ldc, batchCount"
+            "with the bound [val] <= %d", INT_MAX);
+  }
+
```

Comment:
Why does the ifdef span over the entire function? The signature and the error checking look the same. Just ifdef in the body.